docs/GraphRepresentation.md

```diff
@@ -0,0 +1,321 @@
+    In particular, some or all pairs of vertices may be connected by edges
+    running in both directions but bearing different weights.
+- Both vertex in-degree and out-degree may take values from 0 to $|V|$,
+    and may be different.
+    In general, a treated vertex may have any number of incident edges.
+- Edge weights represent general relevance scores, which may not be symmetric.
+    The weight on edge (A,B) represents the relevance of B in relation to A,
+    which may be different from the relevance of A in relation to B.
+- It may or may not be strongly or weakly connected.
+
+
+### Recommendation graph
+
+The recommendation graph is obtained by dropping all edges except those
+leading to each add-on's top N most relevant recommendations,
+with these properties:
+
+- It is directed.
+    For example, that B is a recommendation for A does not imply that
+    A is a recommendation for B.
+- Vertex in-degree may take values from 0 to $|V|$,
+    meaning that any number of add-ons may recommend the current vertex add-on.
+    Vertex out-degree is constrained to be at most N, representing the add-ons
+    recommended for the current vertex add-on.
+- Edge weights no longer play a role.
+    Without loss of generality, they can all be set to 1.
+- It may or may not be strongly or weakly connected.
+    In particular, there may be a number of add-ons which have outgoing edges
+    but no incoming edges,
+    in that they have associated recommendations,
+    but are never recommended from other add-ons.
+
+
+# Treatments
+
+The following [treatments](../taar_lite/recommenders/treatments.py) are implemented for the `GuidGuidCoinstallRecommender`.
+
+
+## Normalizations
+
+One issue with drawing recommendations directly
+from the raw coinstallation graph
+is that popular add-ons can greatly outweigh more relevant add-ons,
+since the distribution of add-on install counts is heavily skewed.
+If a popular add-on such as AdBlock Plus is coinstalled with a given add-on A,
+it will likely have the highest raw count out all add-ons coinstalled with A,
+and thus would top the list of recommendations.
+Furthermore, such popular add-ons are likely to appear
+on most add-ons' coinstall lists.
+Thus, we seek to transform coinstallation counts in such a way as to discount
+the effect of overall popularity.
+This corresponds to adjusting the edge weights in the graph representation.
+
+Note that, even though the raw coinstallation graph is undirected
+(with a symmetric adjacency matrix),
+the graph generated by the normalization treatments may no longer be undirected.
+Although no edges are added or removed,
+the edge weights may be modified asymmetrically,
+meaning that a single undirected edge in the original graph
+may get transformed into two directed edges with different weights
+in the result.
+
+Although these normalizations are described
+in terms of the raw coinstallation matrix,
```

Comment:
I've been using "coinstallation" mainly as a stylistic choice to avoid having hyphens all over the place. (On the other hand, the official Mozilla term "add-on" is hyphenated, so I've also used that consistently :)).